src/Fixture/AbstractExampleFactoryAwareFixture.php

```diff
@@ -0,0 +1,140 @@
+                ->setNormalizer('custom', function (Options $options, array $custom) {
+                    if ($options['random'] <= 0) {
+                        return $custom;
+                    }
+
+                    return array_merge($custom, array_fill(0, $options['random'], $options['prototype']));
+                })
+        ;
+    }
+
+    final public function load(array $options): void
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        $i = 0;
+        foreach ($options['custom'] as $resourceOptions) {
+            $resource = $this->getExampleFactory()->create($resourceOptions);
+
+            $this->objectManager->persist($resource);
+
+            ++$i;
+
+            if (0 === ($i % 10)) {
+                $this->objectManager->flush();
+                $this->objectManager->clear();
+            }
+        }
+
+        $this->objectManager->flush();
+        $this->objectManager->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder($this->getName());
+
+        /** @var ArrayNodeDefinition $optionsNode */
+        $optionsNode = $treeBuilder->getRootNode();
+
+        $nodeBuilder = $optionsNode->children();
+        $nodeBuilder->integerNode('random')->min(0)->defaultValue(0)->end();
+        $nodeBuilder->variableNode('prototype')->end();
+
+        /** @var ArrayNodeDefinition $resourcesNode */
+        $resourcesNode = $optionsNode->children()->arrayNode('custom');
+
+        /** @var ArrayNodeDefinition $resourceNode */
+        $resourceNode = $resourcesNode->requiresAtLeastOneElement()->arrayPrototype();
+        $this->configureResourceNode($resourceNode);
+
+        return $treeBuilder;
+    }
+
+    protected function getExampleFactory(): ExampleFactoryInterface
+    {
+        if (null === $this->exampleFactory) {
+            $exampleFactory = $this->createExampleFactory();
+
+            if (null === $exampleFactory) {
+                throw new \LogicException(sprintf('No example Factory class was found. Use %s or override getExampleFactoryClass method',$this->getExampleFactoryClass()));
+            }
+
+            if (!$exampleFactory instanceof ExampleFactoryInterface) {
+                throw new \LogicException(sprintf('Example Factory "%s" must implement %s.', \get_class($exampleFactory), ExampleFactoryInterface::class));
+            }
+
+            $this->exampleFactory = $exampleFactory;
+        }
+
+        return $this->exampleFactory;
+    }
+
+    protected function getExampleFactoryClass(): string
+    {
+        $names = explode('\\', static::class);
+        $name = end($names);
+        $basename = preg_replace('/(.+)Fixture$/', '${1}', $name);
+
+        $pos = strrpos(static::class, '\\');
+        $namespace = false === $pos ? '' : substr(static::class, 0, $pos);
+
+        return $namespace.'\\Factory\\'.$basename.'ExampleFactory';
```

Comment:
@lchrusciel Maybe can we use $namespace.$basename.'ExampleFactory' instead to have a simpler convention?